contrib/vdirsyncer/calcurse-vdirsyncer.py

```diff
@@ -0,0 +1,150 @@
+
+
+def check_directory(directory):
+    """Check if a directory exists"""
+    if not os.path.isdir(directory):
+        sys.exit("Error. Invalid directory {0}".format(directory))
+
+
+def write_file(file, contents):
+    """Write to file"""
+    print("Writing to file {0}".format(file))
+    with open(file, 'w') as f:
+        f.write(contents)
+
+
+def remove_file(file):
+    """Remove file"""
+    if os.path.isfile(file):
+        print("Deleting file {0}".format(file))
+        os.remove(file)
+
+
+def calcurse_export():
+    """Return raw calcurse data"""
+    command = calcurse + ['-xical', '--export-uid']
+    proc = subprocess.Popen(command, stdout=subprocess.PIPE)
+    return [x for x in io.TextIOWrapper(proc.stdout, encoding="utf-8")]
+
+
+def calcurse_remove(uid):
+    """Remove calcurse event by uid"""
+    command = calcurse + ['-P', '--filter-hash=' + uid]
+    subprocess.call(command)
+
+
+def calcurse_import(file):
+    """Import ics file to calcurse"""
+    command = calcurse + ['-i', file]
+    subprocess.call(command)
+
+<<<<<<< HEAD
+def calcurse_list():
+    """Return all calcurse item uids"""
+    command = calcurse + [
+        '-G',
+        '--format-apt=%(hash)\\n',
+        '--format-recur-apt=%(hash)\\n',
+        '--format-event=%(hash)\\n',
+        '--format-recur-event=%(hash)\\n',
+        '--format-todo=%(hash)\\n'
+    ]
+    proc = subprocess.Popen(command, stdout=subprocess.PIPE)
+    return [x.strip() for x in io.TextIOWrapper(proc.stdout, encoding="utf-8")]
+
+
+def vdirsyncer_sync():
+    """Sync vdir using vdirsyncer"""
+    command = vdirsyncer + ['sync', '--force-delete']
+    subprocess.call(command)
+
+
+def parse_calcurse_data(raw):
+    """Parse raw calcurse data to a uid/ical dictionary"""
+    header = ''.join(raw[:3])
+    events = re.findall('(BEGIN:.*?END:.*?.\n)', ''.join(raw[3:]), re.DOTALL)
+
+    items = {}
+
+    for item in events:
+        uid = re.findall('UID:(.*?)\n', item)[0]
+        items[uid] = header + item + "END:VCALENDAR\n"
+
+    return items
+
+
+def calcurse_to_vdir():
+    """Export calcurse data to vdir"""
+    raw_events = calcurse_export()
+    events = parse_calcurse_data(raw_events)
+
+    files_old = [vdir + x for x in os.listdir(vdir)]
+    files_new = [vdir + uid + ".ics" for uid in events]
+
+    for file in files_old:
+        if file not in files_new:
+            remove_file(file)
+
+    for uid, event in events.items():
+        file = vdir + uid + ".ics"
+        if file not in files_old:
+            write_file(file, event)
+
+
+def vdir_to_calcurse():
+    """Import vdir data to calcurse"""
+    files_old = [vdir + x + '.ics' for x in calcurse_list()]
+    files_new = [vdir + x for x in os.listdir(vdir)]
+
+    for file in files_new:
+        if file not in files_old:
+            calcurse_import(file)
+
+    for file in files_old:
+        if file not in files_new:
+            calcurse_remove(file.replace(vdir, "").replace(".ics", ""))
```

Comment:
Again, I am sure there is a better way to do this. Why don't you drop the `vdir` prefix from `files_old` and `files_new`? You can then simply write `calcurse_import(vdir + file)` above and do not need to do the first string replacement here.

Maybe we can also turn `for x in os.listdir(vdir)` above into `for x in os.listdir(vdir) if x.endswith('.ics')`? Just to be safe.

And maybe use `[:-4]` instead of `.replace(".ics", "")`.

Reply:
Done, thank you for the suggestions.